phypayload.go

```diff
@@ -288,6 +288,20 @@ func (p PHYPayload) ValidateMIC(key AES128Key) (bool, error) {
 	return true, nil
 }
 
+// DevAddr returns a device address (if any) associated to payload
+func (p PHYPayload) DevAddr() (DevAddr, error) {
+	if p.MACPayload == nil {
+		return DevAddr{}, errors.New("lorawan: MACPayload should not be empty")
+	}
+
+	macpayload, ok := p.MACPayload.(*MACPayload)
+	if !ok {
+		return DevAddr{}, errors.New("lorawan: unable to get address of a join message")
```

Comment:
Note that join related payloads are either ``JoinRequestPayload`` or ``JoinAcceptPayload``. Based on the ``MType`` you are able to know to which type you need to cast the ``p.MACPayload``.